src/mlpack/core/tree/rectangle_tree/minimal_coverage_sweep.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file minimal_coverage_sweep.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Definition of the MinimalCoverageSweep class, a class that finds a partition
+ * of a node along an axis.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_MINIMAL_COVERAGE_SWEEP_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_MINIMAL_COVERAGE_SWEEP_HPP
+
+namespace mlpack {
+namespace tree {
+
+constexpr double fillFactor = 0.5;
+
+/**
+ * The MinimalCoverageSweep class finds a partition along which we
+ * can split a node according to the coverage of two resulting nodes.
+ * Moreover, the class evaluates the cost of each split.
+ *
+ * @tparam SplitPolicy The class that provides rules for inserting children of
+ *    a node that is being split into two new subtrees.
+ */
+template<typename SplitPolicy>
+class MinimalCoverageSweep
+{
+ private:
+  /**
+   * Class to allow for faster sorting.
+   */
+  template<typename ElemType>
+  struct SortStruct
+  {
+    ElemType d;
+    int n;
+  };
```

Comment:
Okay, I opened #712 about this and we can figure it out there instead of inside this PR, so we can get this one done more quickly. It is probably easier to do comparisons by making a separate branch from this one. Don't feel obligated to do that; I will eventually get around to it if you don't want to. But it seems like you never turn things down, so I guess you will probably do it anyway. :)